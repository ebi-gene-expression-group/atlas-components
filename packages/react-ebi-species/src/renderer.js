"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
