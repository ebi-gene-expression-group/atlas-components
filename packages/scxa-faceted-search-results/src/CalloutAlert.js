import React from 'react'
import PropTypes from 'prop-types'

const CalloutAlert = ({error}) =>
  <div className={`row column`}>
    <div className={`callout alert small`}>
      <h5>Oops!</h5>
      <p>
        {error.description}<br/>
        If the error persists, in order to help us debug the issue, please copy the URL and this message and
        send it to us via <a href={`https://www.ebi.ac.uk/support/gxasc`}>the EBI Support & Feedback system</a>:
      </p>
      <code>{`${error.name}: ${error.message}`}</code>
    </div>
  </div>

CalloutAlert.propTypes = {
  error: PropTypes.shape({
    description: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    message: PropTypes.string.isRequired
  })
}

export default CalloutAlert
