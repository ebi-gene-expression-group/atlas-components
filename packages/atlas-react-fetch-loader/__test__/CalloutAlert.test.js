import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import CalloutAlert from '../src/CalloutAlert'

describe(`CalloutAlert`, () => {
  const props = {
    error: {
      description: `A human-readable description of the error, hopefully useful to the user`,
      name: `Error name`,
      message: `Error message`
    }
  }

  it(`prints all the relevant error information`, () => {
    const wrapper = shallow(<CalloutAlert {...props} />)
    expect(wrapper).toIncludeText(props.error.description)
    expect(wrapper).toIncludeText(props.error.name)
    expect(wrapper).toIncludeText(props.error.message)
  })

  it(`matches snapshot`, () => {
    const tree = renderer.create(<CalloutAlert {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
