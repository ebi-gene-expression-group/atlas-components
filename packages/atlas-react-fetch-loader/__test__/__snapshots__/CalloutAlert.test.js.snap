// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalloutAlert matches snapshot 1`] = `
<div
  className="row column"
>
  <div
    className="callout alert small"
  >
    <h5>
      Oops!
    </h5>
    <p>
      A human-readable description of the error, hopefully useful to the user
      <br />
      If the error persists, in order to help us debug the issue, please copy the URL and this message and send it to us via 
      <a
        href="https://www.ebi.ac.uk/support/gxasc"
      >
        the EBI Support & Feedback system
      </a>
      :
    </p>
    <code>
      Error name: Error message
    </code>
  </div>
</div>
`;
